packages/app/src/cli/services/context/partner-account-info.ts

```diff
@@ -0,0 +1,76 @@
+import {geCurrentAccountInfo} from '../../api/graphql/current_account_info.js'
+import {outputDebug} from '@shopify/cli-kit/node/output'
+import {ensureAuthenticatedPartners} from '@shopify/cli-kit/node/session'
+
+export interface PartnersSession {
+  token: string
+  accountInfo: AccountInfo
+}
+
+export abstract class AccountInfo {
+  isUserAccount(): this is UserAccountInfo {
+    return false
+  }
+
+  isServiceAccount(): this is ServiceAccountInfo {
+    return false
+  }
+
+  isUnknownAccount(): this is UnknownAccountInfo {
+    return false
+  }
+}
+
+export class UserAccountInfo extends AccountInfo {
```

Comment:
In this case I have done this to avoid add casting in places where the `email`  or `orgName` is needed and having a plain object with lots of optional values.

https://github.com/Shopify/cli/blob/dde2167f912163cb08d5b9e6303642832a4d6cc9/packages/app/src/cli/services/info.ts#L99:L105

An alternative could be having a plain object with `type` and two optional fields 'orgName' and 'email', and check on the `type` to trigger different logics

Reply:
I'll modify this to use types/interfaces and functions to be consistent with the rest of the code

Reply:
Oh no need. if you think this is the better approach, I can understand that, I was just curious